python/src/dataset.rs

```diff
@@ -597,10 +599,11 @@ impl Dataset {
         columns: Vec<&str>,
         index_type: &str,
         name: Option<String>,
-        metric_type: Option<&str>,
+        replace: Option<bool>,
```

Comment:
I moved `replace` out of `kwargs` and moved `metric_type` into `kwargs` since:

 * `replace` is universally applicable regardless of index type
 * `metric_type` is only applicable for vector indices